### DISCUSSION TITLE
Forecasting time series using SVR

### DISCUSSION THREAD

**Discussion by akmalmvic0312** [alan-turing-institute/sktime#5195] [Q&A]
Hello! I've a question about scaling data in sktime. I want to make a forecasting model using SVR which is need some tools such as make reduction, scaling or mlflow to track the parameter output. 
1. If i use scaling data in pipeline or _TransformedTargetForecaster_, i should to inverse the output or no?
2. If i create 2 file .py, _ml_train.py_ to train the model SVR forecasting (using _best_model = gscv.save(path=model_output)_ to save the best model) and _ml_forecast.py_ to load the best model from _ml_train.py_ and just _.predict(fh)_ using this model. Should i inverse the predict output on _ml_forecast.py_ using scaler in _ml_train.py_?

Here is the Python script:

```
# Create dataset for training and validation (80% for training)
freq = 'H'
# Test size
test_size = 48
# Split data
train, test = temporal_train_test_split(df_1, test_size=test_size)

# Scale data (standarized data)
train_scaled_df, test_scaled_df, scaler = helper.data_transformation(train, test)

# Horizon
fh = ForecastingHorizon(test_scaled_df.index, is_relative=False)
fh_int = np.arange(1, len(fh))
fh_2 = np.arange(1, 25)
window_length = 48

loginfo = "Features dimension: {}".format(train_scaled_df.shape)
loggingInfo(loginfo)

loginfo = "Target dimension: {}".format(test_scaled_df.shape)
loggingInfo(loginfo)
print("")

###################################################################################
## Train SVR Model
###################################################################################
from sklearn import svm
from sktime.forecasting.model_evaluation import evaluate

with mlflow.start_run(run_name=model_name,
                      description="Training model for each sub system in PHM") as run:

    # Log test size parameter to mlflow
    mlflow.log_param('test_size', test_size)

    C_range = np.logspace(-4, 3, 20)
    gamma_range = np.logspace(-4, 3, 20)
    kernel_range = ['rbf', 'linear']
    param_grid = {"estimator__gamma": gamma_range, "estimator__C": C_range, "estimator__kernel": kernel_range}

    print("")
    loginfo = "Creating model ......."
    loggingInfo(loginfo)
    t0 = time.process_time()

    # Perform Grid search with KFold cross validation to find optimal hyperparameter
    forecaster = make_reduction(svm.SVR(), strategy="recursive")
    # pipe = TransformedTargetForecaster(steps=[
    #     ("imputer", Imputer(method="mean")),
    #     ("forecaster", make_reduction(svm.SVR(), window_length=48, strategy="recursive")),
    # ])

    cv = SingleWindowSplitter(fh=fh_int)
    #cv = ExpandingWindowSplitter(fh=fh_2, initial_window=24, step_length=1)
    #cv = SlidingWindowSplitter(fh=fh_2, window_length=window_length, step_length=7)

    gscv = ForecastingGridSearchCV(
        forecaster=forecaster,
        param_grid=param_grid,
        cv=cv,
        n_jobs=-1,
        verbose=1,
        scoring=MeanAbsolutePercentageError(symmetric=True),  # set custom scoring function
        error_score='raise',
        refit=True
    )
    
    # Make ML Model
    gscv.fit(train_scaled_df)

    # Get best model from grid search
    best_model = gscv.save(path=model_output)
```

**Comment by hazrulakmal** [comment]
Hi @akmalmvic0312, I suppose this is the type of question that should be raised in the discussions section, not here. Issues are used to track bugs, enhancements, and matters related to sktime development. 

Anyway, regarding your question.
>If i use scaling data in pipeline or TransformedTargetForecaster, i should to inverse the output or no?

No. you don't have to. `TransformedTargetForecaster` handles both preprocessing (before fitting, i.e `.fit`) and postprocessing (after prediction `.predict`, i.e the inverse) under the hood. 

>Should i inverse the predict output on ml_forecast.py using scaler in ml_train.py?

Since, you save `ForecastingGridSearchCV` object and when you load it back, it should be in the same state so you don't have to do the inverse, `TransformedTargetForecaster` should handle it.

**Comment by akmalmvic0312** [comment]
Hi, @hazrulakmal !

Yes, i was open disscuse in discord channel about that, but i write in here because i want to share my python script. Sorry in the above python script, i comment my TransformedTargetForecaster because i still confused about this pipeline, how inverse scaling data after .predict() or scaling data after i load it back. But now i was done about this issue. Thank you!

**Comment by hazrulakmal** [comment]
Sorry, I meant GitHub discussion section [here](https://github.com/sktime/sktime/discussions), not discord discussion channel. No worries. If you think you are done with this issue, could you kindly close it? Thanks!

**Reply by fkiraly** [reply]
@hazrulakmal, it's possile to convert issues to discussions, I did just that - no need to close.
